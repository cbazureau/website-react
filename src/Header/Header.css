.Header {
  z-index: 1;
  height: 350px;
  overflow: hidden;
}

.Header__background {
  height: 100vh;
  object-fit: cover;
  width: 100%;
  position: fixed;
}

.Header::after {
  position: fixed;
  content: '';
  height: 100vh;
  width: 100%;
  z-index: 2;
  background: linear-gradient(to bottom, transparent, transparent 60%, var(--neutral-light) 80%);
}

@media only screen and (min-width: 620px) {
  .Header {
    height: 500px;
  }
}

.Header__txt {
  position: absolute;
  color: #fff;
  padding: 8em 0 6em;
  display: flex;
  flex-direction: column;
  align-items: center;
  z-index: 3;
  width: 100%;
}

.Header__banner {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Header__logo {
  border-radius: 50%;
  width: 100px;
  display: block;
  height: 100px;
}

.Header__name {
  font-size: 3.4rem;
  font-weight: normal;
}

.Header__job {
  margin-top: 0;
  font-size: 2rem;
}

@media only screen and (min-width: 620px) {
  .Header__banner {
    padding: 4em 0 0;
  }
}
