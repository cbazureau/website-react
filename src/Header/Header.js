import React from 'react';
import Menu from './Menu/Menu';
import Image from '../Image/Image';
import './Header.css';

const Header = ({ onSectionChange }) => (
  <header className="Header">
    <Image
      className="Header__background"
      src="/static/img/banner.jpg"
      alt="banner background"
      isLazyLoaded={false}
    />
    <div className="Header__txt">
      <Menu onSectionChange={onSectionChange} />
      <div className="Header__banner">
        <Image
          className="Header__logo"
          src="/static/img/identite.jpg"
          alt="logo"
          isLazyLoaded={false}
        />
        <h1 className="Header__name">C&eacute;dric Bazureau</h1>
        <h2 className="Header__job" data-section="headline">
          Lead Tech Front Ã  Renault
        </h2>
      </div>
    </div>
  </header>
);

export default Header;
