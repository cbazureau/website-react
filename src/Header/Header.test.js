import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

jest.mock('./Menu/Menu', () => 'menu');
jest.mock('../Image/Image', () => 'image');

const onSectionChange = () => () => null;

describe('Header component', () => {
  it('renders with props', () => {
    const component = renderer.create(<Header onSectionChange={onSectionChange} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
