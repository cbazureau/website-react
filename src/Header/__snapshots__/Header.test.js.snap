// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders with props 1`] = `
<header
  className="Header"
>
  <image
    alt="banner background"
    className="Header__background"
    isLazyLoaded={false}
    isRetina={false}
    src="/static/img/banner.jpg"
  />
  <div
    className="Header__txt"
  >
    <menu
      onSectionChange={[Function]}
    />
    <div
      className="Header__banner"
    >
      <image
        alt="logo"
        className="Header__logo"
        height="100px"
        isLazyLoaded={false}
        src="/static/img/identite.jpg"
        width="100px"
      />
      <h1
        className="Header__name"
      >
        Cédric Bazureau
      </h1>
      <h2
        className="Header__job"
        data-section="headline"
      >
        Lead Tech Front à Renault
      </h2>
    </div>
  </div>
</header>
`;

exports[`Header component renders without crashing 1`] = `
<header
  className="Header"
>
  <image
    alt="banner background"
    className="Header__background"
    isLazyLoaded={false}
    isRetina={false}
    src="/static/img/banner.jpg"
  />
  <div
    className="Header__txt"
  >
    <menu />
    <div
      className="Header__banner"
    >
      <image
        alt="logo"
        className="Header__logo"
        height="100px"
        isLazyLoaded={false}
        src="/static/img/identite.jpg"
        width="100px"
      />
      <h1
        className="Header__name"
      >
        Cédric Bazureau
      </h1>
      <h2
        className="Header__job"
        data-section="headline"
      >
        Lead Tech Front à Renault
      </h2>
    </div>
  </div>
</header>
`;
